collect_app/src/main/java/org/odk/collect/android/geo/MapProvider.java

```diff
@@ -31,13 +33,12 @@
 import static org.odk.collect.android.preferences.GeneralKeys.KEY_USGS_MAP_STYLE;
 
 /**
- * A static class that obtains a MapFragment according to the user's preferences.
+ * Obtains a MapFragment according to the user's preferences.
  * This is the top-level class that should be used by the rest of the application.
  * The available options on the Maps preferences screen are also defined here.
  */
+@Singleton
```

Comment:
You don't need the `@Singleton` here as you have it on the provider. It can be useful as documentation but I'm not sure the class needs to know it's going to be used as a singleton in this case.

Reply:
I went back and forth on this and it seemed helpful to identify that the map fields are an ongoing concern. How do you decide whether it’s useful or not?

Reply:
Hmmm. I guess for me I like that all the configuration around how a dependency is provided lives in the Dagger files (`AppDependencyModule` for instance) and thatt really the object itelf shouldn't need to know about it's own lifecycle. 

A more concrete criticism would be that having `@Singleton` in both places is confusing for newbies to Dagger (but then what about Dagger isn't confusing).

Reply:
> thatt really the object itelf shouldn't need to know about it's own lifecycle.

I see what you're getting at. I feel like singletons often have a pretty different structure and the assumption that a single instance exists is baked in but perhaps that shouldn't be the case. Here, like I wrote in the PR description, it could pretty easily be restructured to not be a singleton. Anyway, I don't feel strongly about it so I've removed it.

> having @Singleton in both places is confusing for newbies to Dagger
I do agree with that.